### PR TITLE
cleanup `replaceValInIndexVal`

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Rename `replaceValInIndexVal` into `replaceValInDef`, make it support all expr types.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

File: csrc/ir/utils.h [hunk 1]
```diff
@@ -173,16 +173,15 @@ TORCH_CUDA_CU_API Expr* replaceValInExpr(
...
+//! Replace Vals in the definition of the given Val as specified by
+//! replacement_map while cloning the given Val.
 //!
 //! This is similar to replaceValInExpr but is different as Vals are
 //! cloned such that no other exprs using the same leaf Vals are not
 //! modified. TODO: Consider cleaning up the multiple replacement
 //! routines.
-Val* replaceValInIndexVal(
-    Val* index,
+Val* replaceValInDef(
```
Comment (naoyam) [inline csrc/ir/utils.h:183]: nit: the naming might be a little confusing as the replacement is not necessarily done in the definition. Maybe, something like `replaceValRecursively`?

**Review by naoyam — APPROVED** [review]
LGTM

File: csrc/ir/utils.h (hunk 1 above)
Comment (zasdfgbnm) [reply csrc/ir/utils.h]: changed

**Pull request merged**